weaviate/client.py

```diff
@@ -2,39 +2,35 @@
 Client class definition.
 """
 
-import asyncio
 from typing import Optional, Tuple, Union, Any
 
 from typing_extensions import deprecated
 
-from weaviate import syncify
-from weaviate.backup.backup import _BackupAsync
-from weaviate.backup.sync import _Backup
-from weaviate.event_loop import _EventLoopSingleton, _EventLoop
 
-from weaviate.users.users import _UsersAsync
-
-from weaviate.users.sync import _Users
 from .auth import AuthCredentials
+from .backup import _BackupAsync, _Backup
 from .client_base import _WeaviateClientBase
 from .collections.batch.client import _BatchClientWrapper
 from .collections.cluster import _Cluster, _ClusterAsync
 from .collections.collections.async_ import _CollectionsAsync
 from .collections.collections.sync import _Collections
 from .config import AdditionalConfig
+from .connect import executor
 from .connect.base import (
     ConnectionParams,
 )
+from .connect.v4 import ConnectionAsync, ConnectionSync
 from .debug import _Debug, _DebugAsync
 from .embedded import EmbeddedOptions
 from .rbac import _RolesAsync, _Roles
 from .types import NUMBER
+from .users import _UsersAsync, _Users
 
 TIMEOUT_TYPE = Union[Tuple[NUMBER, NUMBER], NUMBER]
 
 
-@syncify.convert
-class WeaviateClient(_WeaviateClientBase):
+@executor.wrap("async")
```

Comment:
why do we need the wrapper here?

Reply:
because the clients inherit from `_WeaviateClientExecutor` that contains I/O methods returning `executor.Result` values